firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileDownloadService.java

```diff
@@ -138,18 +162,30 @@ public static ModelFileDownloadService getInstance() {
     return registerReceiverForDownloadId(newDownloadId, customModel.getName());
   }
 
+  /** Removes or cancels the downloading model if exists. */
+  synchronized void removeOrCancelDownload(String modelName, Long downloadId) {
+    if (downloadManager != null && downloadId != 0) {
+      downloadManager.remove(downloadId);
+    }
+    sharedPreferencesUtil.setFailedUploadedCustomModelDetails(modelName);
```

Comment:
From reading this function I get the impression that we only removeOrCancel failed uploaded custom models. Is that right?

Reply:
Also immediately after a successful download has it's file transferred ( after moveModelToDestinationFolder is called)

Reply:
Got it. The `setFailedUploadedCustomModelDetails` seems to imply that it's for failure only.